Framework/CurveFitting/src/Functions/DecoupAsymPowderMagLong.cpp

```diff
@@ -45,10 +45,13 @@ void DecoupAsymPowderMagLong::functionDeriv1D(Jacobian *out, const double *xValu
 
   for (size_t i = 0; i < nData; i++) {
     double diffasym = getAz(xValues[i], charField);
-    double diffcharfield = -xValues[i] / pow(charField, 2);
-    double b = xValues[i] / charField;
-    double diffb = ((pow(b, 2) - 1) * ((pow(b, 2) + 3) * log((b + 1.0) / (b - 1.0)) - (2 * b))) / (8 * pow(b, 4));
-    double diffAz = diffcharfield * diffb;
+    double diffAz = getDiffAz(xValues[i], charField);
+    if (!std::isfinite(diffasym)) {
+      diffasym = 0;
```

Comment:
should these just be large numbers instead of 0? Since the condition is that it has become non-finite? It does have the question of making sure that the large number has the right sign (positive or negative)

Reply:
zero is correct in the limit of b->0 https://www.wolframalpha.com/input/?i=lim+x-%3E0+log%28%28x%2B1%29%2F%28x-1%29%29%2Fx%5E4